### ISSUE TITLE
Interactive marker doesn't appear if timestamp is set if rviz fixed frame is on different frame

### ISSUE THREAD

**Issue by lucasw** [ros-visualization/interactive_markers#34]
If an interactive marker stamp is set then the marker doesn't appear, unless rviz fixed frame is on the same frame (moving_frame in the example below).

```
def makeMovingMarker(position):
    int_marker = InteractiveMarker()
    int_marker.header.frame_id = "moving_frame"
    int_marker.header.stamp = rospy.Time.now()
...
```
https://github.com/ros-visualization/visualization_tutorials/blob/kinetic-devel/interactive_marker_tutorials/scripts/basic_controls.py#L394

http://answers.ros.org/question/262866/interactive-marker-attached-to-a-moving-frame/

**Comment by dgossow** [comment]
Does the InteractiveMarker display show an error? I could imagine that it tries to query the transform from "moving_frame" to whatever the fixed frame is in RViz, but that's not available yet

**Comment by lucasw** [comment]
The status shows ok.

The main problem mogumbo brought up and I also observe is that all the markers from the same server also don't appear.

An additional detail is that if I run basic_controls.py without setting the timestamp (which is the same as rospy.Time(0) I think) and have rviz showing it, kill basic_controls.py, add stamp = rospy.Time.now(), then rerun basic_controls the moving frame marker then appears frozen (presumably where it should) and the rest of the markers also are visible.

Except for all the other markers not showing up this makes sense (though probably should still generate a warning): basic_controls.py hasn't sent the moving tf frame when rospy.Time.now() is called so there is nothing to look up, unless the basic_controls.py was running recently in the past.

**Comment by lucasw** [comment]
I now see the rviz interactive marker display is in https://github.com/ros-visualization/rviz ( https://github.com/ros-visualization/rviz/tree/kinetic-devel/src/rviz/default_plugin/interactive_markers ) and not here, the problem lies there.

**Comment by rhaschke** [comment]
I was able to reproduce the issue with the python version (but not the cpp version) of basic_controls.
I tracked it down to a timing issue: In python the tf transform published by the example code, is published later than the interactive marker message is assembled. Thus the timestamp of the tf transform is older than the timestamp of the marker message, resulting in rviz postponing the init message (waiting for a newer update). Unfortunately, the corresponding warning message was overridden by another, later status message. If fixed this in 0fb73f86c462f1fa87b938346b4d5e40f18b34f1.

That the moving marker is not updated is desired as the marker is not re-published with a more recent time stamp.
